tests/test_jacobi.c

```diff
@@ -0,0 +1,99 @@
+#include "../src/jacobi.h"
+#include "stdio.h"
+
+#define PI 3.141592653589793238
+#define EPSILON 0.001
+
+int testMaximumIndex(real_number* input, vec_size size, vec_size expectedRow, vec_size expectedCol) {
+
+    vec_size maxRow;
+    vec_size maxCol;
+    jacobiMaxIndex(input, size, &maxRow, &maxCol);
+
+    if (maxRow == expectedRow && maxCol == expectedCol) {
+        printf("Success : Test maximum index\n");
+        return 0;
+    } else {
+        printf("Fail : Test maximum index. Row should be 1 and column should be 0, but row is %d and column is %d\n", maxRow, maxCol);
+        return 1;
+    }
+}
+
+int testJacobi(real_number* input, real_number* expectedEigenValues, real_number* expectedEigenVectors, vec_size size, vec_size iterations, vec_size cyclic) {
+
+    vec_size squaredSize = size * size;
+
+    real_number mat1[squaredSize];
+    memcpy(mat1, input, sizeof(real_number) * squaredSize);
+
+    real_number output[squaredSize];
+    memset(output, 0, sizeof(real_number) * squaredSize);
```

Comment:
Oh, that's interesting. We have to memset the output matrix. Is that written somewhere in the documentation? I don't remember seeing that?

Reply:
we don't have too, I removed it